### DISCUSSION TITLE
Problem building the Navier-Stokes module after update

### DISCUSSION THREAD

**Discussion by Eilloo** [idaholab/moose#25896] [Q&A Getting Started]
Hi all,

I'd not updated my MOOSE install for a while and figured it was about time. After following the steps for staying up-to-date [here](https://mooseframework.inl.gov/getting_started/new_users.html), I encountered a series of errors when trying to build any app with the Navier Stokes module enabled in the Makefile. The errors all looked something like this:

```
In file included from /Users/opaice/Documents/moose_projects/moose/modules/navier_stokes/build/unity_src/executioners_Unity.C:2:
/Users/opaice/Documents/moose_projects/moose/modules/navier_stokes/src/executioners/SIMPLE.C:929:19: error: no matching member function for call to 'solve'
  pressure_solver.solve(mmat, mmat, solution, rhs);
  ~~~~~~~~~~~~~~~~^~~~~
/Users/opaice/mambaforge3/envs/moose/libmesh/include/libmesh/petsc_linear_solver.h:155:3: note: candidate function not viable: requires 5 arguments, but 4 were provided
  solve (SparseMatrix<T> & matrix_in,
  ^
```

To be sure it wasn't anything in my existing apps causing the issues, I created a new app just test this and the problem persisted.
I also tried building in a 'vanilla' environment as described [here](), which also didn't solve the problem.

The line where `pressure_solver` is defined is:
```
  // Fetch the linear solver from the system
  PetscLinearSolver<Real> & pressure_solver = libMesh::cast_ref<PetscLinearSolver<Real> &>(*pressure_system.get_linear_solver());
```

So it seems that whichever solver is being fetched is mismatched with the arguments we later try and give to it. I wonder if I somehow have the wrong version of petsc installed?

I am curious whether anyone else sees this behaviour or if it's something I've done wrong in the update. 

Any ideas are welcome - thanks!

**Comment by GiudGiud** [comment]
Hello

You just need to update libmesh
We're working on changing the package right right now so it may not work right right now
I ll let you know once the package is fixed

Guillaume

**Reply by Eilloo** [reply]
Ah, good to know - that would be great, thanks!
